### PR TITLE
feat: Rstar2 update

### PR DESCRIPTION
**Pull request by nufukim**
rstar2-agent RL training recipe

### Description

fix issue [support rstar2-agent](https://github.com/RLinf/RLinf/issues/78)
I implemented the simple rstar2-agent algorithm.
- agentic rollout with the Python execution environment:
    A simple data distribution method is used, without involving load balancing. 
    Use sglang as rollout backend.

- fused multi- math verification tools during reward stage:
    It can be called correctly now

- add rollout trace data augmentation stage after rollout & reward stage
    The sampling logic has been implemented.

- align training hyperparameters
    TODO:...

### Motivation and Context
link: [paper](https://www.arxiv.org/pdf/2508.20722).

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: examples/rstar2/agent_loop/rstar2_agent_loop.py [hunk 1]
```diff
@@ -0,0 +1,287 @@
+# Copyright 2025 The RLinf Authors.
```
Comment (qurakchin) [inline examples/rstar2/agent_loop/rstar2_agent_loop.py]: this file is not needed

File: examples/rstar2/__init__.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2025 The RLinf Authors.
```
Comment (qurakchin) [inline examples/rstar2/__init__.py:1]: this file is not needed.

File: examples/rstar2/tools/__init__.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2025 The RLinf Authors.
```
Comment (qurakchin) [inline examples/rstar2/tools/__init__.py:1]: this file is not needed.

File: rlinf/agents/rstar2/rstar2_agent_loop.py [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+)
+from rlinf.scheduler import Channel
+from rlinf.utils.placement import ModelParallelComponentPlacement
+from rlinf.workers.agent.agent_loop import AgentLoopOutput, AgentLoopWorker
+
+@dataclass
+class GenerateContext:
+    tool_session_ids: dict[str, str] = field(default_factory=dict)
+
+class FunctionCall(BaseModel):
```
Comment (qurakchin) [inline rlinf/agents/rstar2/rstar2_agent_loop.py]: is FunctionCall used?

File: rlinf/utils/distributed.py [hunk 5]
```diff
@@ -164,10 +164,14 @@ def from_rollout_batches(
     ) -> Self:
         current_device = torch.cuda.current_device()
 
-        attn_mask = rollout_batches.get("attention_mask")
-        current_num_samples = attn_mask.size(0)
-
-        # 2. Calculate local sample token counts
+        # 1. 获取本地样本数
```
Comment (qurakchin) [inline rlinf/utils/distributed.py]: use english comment.

**Review by qurakchin — CHANGES_REQUESTED** [review]

File: rlinf/utils/distributed.py (hunk 5 above)
Comment (nufukim) [reply rlinf/utils/distributed.py]: Done

File: examples/rstar2/tools/__init__.py (hunk 3 above)
Comment (nufukim) [reply examples/rstar2/tools/__init__.py]: Done

File: examples/rstar2/agent_loop/rstar2_agent_loop.py (hunk 1 above)
Comment (nufukim) [reply examples/rstar2/agent_loop/rstar2_agent_loop.py]: Done

File: rlinf/agents/rstar2/rstar2_agent_loop.py (hunk 4 above)
Comment (nufukim) [reply rlinf/agents/rstar2/rstar2_agent_loop.py]: yes

File: examples/rstar2/__init__.py (hunk 2 above)
Comment (nufukim) [reply examples/rstar2/__init__.py:1]: Done

File: rlinf/data/datasets/rstar2.py [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+from omegaconf import OmegaConf, DictConfig
+from torch.utils.data import Dataset
+from transformers import AutoTokenizer
+
+from rlinf.data.datasets.item import DatasetItem
+from rlinf.data.datasets.utils import batch_pad_to_fixed_len
+
+def get_tool_schemas():
+    """
+    从配置文件获取工具模式列表
```
Comment (qurakchin) [inline rlinf/data/datasets/rstar2.py:30]: please use english comments.

File: ray_utils/start_ray.sh [hunk 7]
```diff
@@ -15,7 +15,7 @@ RAY_PORT=${RAY_PORT:-29500}  # Default port for Ray, can be modified if needed
 # Head node startup logic
 if [ "$RANK" -eq 0 ]; then
     # Get local machine IP address (assumed to be intranet IP)
-    IP_ADDRESS=$(hostname -I | awk '{print $1}')
+    IP_ADDRESS=$(hostname -I | awk '{print $NF}')
```
Comment (qurakchin) [inline ray_utils/start_ray.sh:18]: why not use "hostname -i"?

File: rlinf/agents/rstar2/rstar2_agent_loop.py [hunk 8]
```diff
@@ -0,0 +1,302 @@
...
+        trace_uuid = uuid4().hex
+        if self.print_outputs:
+            trace_prints: list[dict] = []
+            trace_prints.append({"uuid": trace_uuid, "prompt": orig_prompt_text})
+        response_mask = []
+        response_logprobs = None
+        if self.return_logprobs:
+            response_logprobs = []
+        user_turns, assistant_turns = 0, 0
+        # 5 is a magic number in this demo.
```
Comment (qurakchin) [inline rlinf/agents/rstar2/rstar2_agent_loop.py:217]: this comment is not needed.

File: rlinf/agents/rstar2/rstar2_agent_loop.py [hunk 9]
```diff
@@ -0,0 +1,302 @@
...
+        while True:
```
Comment (qurakchin) [inline rlinf/agents/rstar2/rstar2_agent_loop.py:218]: add more and appropriate comment to explain the main logic of rstar2 agent.
